src/app/space/create/deployments/services/deployments.service.ts

```diff
@@ -259,6 +278,47 @@ export class DeploymentsService implements OnDestroy {
       );
   }
 
+  getDeploymentTimeConstrainedStats(spaceId: string, applicationId: string, environmentName: string, elapsedTime: number, endTime: number = Date.now()): Observable<TimeConstrainedStats> {
+    const startTime = endTime - elapsedTime;
+    const stats = this.getTimeConstrainedTimeseriesData(spaceId, applicationId, environmentName, startTime, endTime);
+    // Process CPU Stats
+    let cpuStats: Observable<CoresSeries[]> = stats.map((t: MultiTimeseriesData) => t.cores);
+    const cpuQuota = this.getEnvironmentCpuStat(spaceId, environmentName).map((stat: CpuStat) => stat.quota);
+    cpuStats = Observable.combineLatest(cpuStats, cpuQuota, (cpuStats: CoresSeries[], cpuQuota: number) => {
```

Comment:
I'd prefer to avoid reassigning cpuStats and memStats like this. Better to have the initial definition be a different name and make them all const - just easier to follow IMO.